freshmaker/pyxis.py

```diff
@@ -0,0 +1,246 @@
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        if params is None:
+            request_params = {}
+        else:
+            request_params = copy.deepcopy(params)
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest',
+                          'data.bundle_path']
+        request_params['include'] = ','.join(include_fields)
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            request_params['filter'] = f"source_index_container_path=={path}"
+            bundles_per_index_image[path] = self._pagination('operators/bundles',
+                                                             request_params)
+
+        return bundles_per_index_image
+
+    def get_latest_bundles(self, index_images, params={}):
+        """
+        Get latest bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: latest bundle images per channel per index image
+        :rtype: list
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret_bundles = []
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle['channel_name']
+                try:
+                    if channel in bundle_per_channel:
+                        new_ver = bundle['version']
+                        old_ver = bundle_per_channel[channel]['version']
+                        if version.Version(new_ver) > version.Version(old_ver):
+                            bundle_per_channel[channel] = bundle
+                    else:
+                        # Check if the right format of version is used
+                        version.Version(bundle.get('version'))
+                        bundle_per_channel[channel] = bundle
+                # Check if the right format of version is used
+                except version.InvalidVersion as e:
+                    path = bundle.get('bundle_path', 'Unknown bundle path')
+                    log.warning("Other format than SemVer is used in "
+                                "bundle: %s", path)
+                    log.warning(repr(e))
+            ret_bundles.extend(bundle_per_channel.values())
+
+        return ret_bundles
+
+    def get_digests_by_nvrs(self, nvrs, params=None):
```

Comment:
It seems `params` isn't ever used. Could you please remove it? It's best to not add unused code. We can always add it when needed.